packaging/os/yum_repository.py

```diff
@@ -44,23 +51,41 @@
         (bytes/sec) then this option is ignored. Default is C(0) (no bandwidth
         throttling).
   baseurl:
-    required: false
+    required: true
```

Comment:
you can keep it as required=false if you have conditional requirements as in the description, required: true should be identical to the params in argument_spec=dict()